pytorch_lightning/loggers/neptune.py

```diff
@@ -412,8 +421,24 @@ def training_step(self, batch, batch_idx):
         return self.run
 
     @property
+    @rank_zero_experiment
     def run(self) -> Run:
-        return self._run_instance
+        try:
+            if not self._run_instance:
+                self._run_instance = neptune.init(**self._neptune_init_args)
+                self._retrieve_run_data()
+                # make sure that we've log integration version for newly created
+                self._run_instance[_INTEGRATION_VERSION_KEY] = __version__
+
+            return self._run_instance
+        except NeptuneLegacyProjectException as e:
+            raise TypeError(
+                f"""Project {self._project_name} has not been migrated to the new structure.
+                You can still integrate it with the Neptune logger using legacy Python API
+                available as part of neptune-contrib package:
+                  - https://docs-legacy.neptune.ai/integrations/pytorch_lightning.html\n
+                """
```

Comment:
Don't use triple quotes as they include the indentation as leading spaces.

```suggestion
                f"Project {self._project_name} has not been migrated to the new structure."
                " You can still integrate it with the Neptune logger using legacy Python API"
                " available as part of neptune-contrib package:"
                " https://docs-legacy.neptune.ai/integrations/pytorch_lightning.html\n
                "
```